code/modules/mob/living/carbon/human/human_helpers.dm

```diff
@@ -273,52 +273,52 @@
 /mob/living/carbon/human/proc/has_working_organs()
 	var/datum/internal_organ/heart/heart = internal_organs_by_name["heart"]
 
-	if(!heart || heart.organ_status == ORGAN_BROKEN || !has_brain())
+	if(species.species_flags & ROBOTIC_LIMBS)
+		return TRUE // combat robots and synthetics don't really have any of these
+	if(!has_brain())
+		return FALSE
+	if(!heart || heart.organ_status == ORGAN_BROKEN)
 		return FALSE
-
 	return TRUE
 
 /**
  * Proc that brings a carbon human back to life. Only works if their health is higher than their death threshold and they are dead in the first place.
  *
- * intended to be called by defibrillators or anything that brings a carbon human back to life
+ * Intended to be called by defibrillators or anything that brings a carbon human back to life
  */
 /mob/living/carbon/human/proc/resuscitate()
-	if(stat == DEAD && health >= get_death_threshold())
-		set_stat(UNCONSCIOUS)
-		emote("gasp")
-		chestburst = CARBON_NO_CHEST_BURST
-		regenerate_icons()
-		reload_fullscreens()
-		flash_act()
-		apply_effect(10, EYE_BLUR)
-		apply_effect(20 SECONDS, PARALYZE)
-		handle_regular_hud_updates()
-		updatehealth() //One more time, so it doesn't show the target as dead on HUDs
-		dead_ticks = 0 //We reset the DNR timer
+	if(stat != DEAD || health <= get_death_threshold())
+		return
+	set_stat(UNCONSCIOUS)
+	chestburst = CARBON_NO_CHEST_BURST
+	regenerate_icons()
+	reload_fullscreens()
+	handle_regular_hud_updates()
+	updatehealth() //One more time, so it doesn't show the target as dead on HUDs
+	dead_ticks = 0 //We reset the DNR timer
 
 /**
  * Proc for checking parameters of a human for defibrillation.
  *
- * Checks decapitation, DNR status, mind/ghost status, damage, organs and having a client for defibrillation.
+ * Checks decapitation, DNR status, organ damage and health status (in that order) for defibrillation.
  *
  * See defines in `__DEFINES/defibrillator.dm` for bitflags.
  *
- * `additional_damage` can be used to add additional damage when calculating health for situations like grabbing ghost.
+ * `additional_health_increase` can be used to add additional health when calculating health for situations like grabbing ghost.
  */
-/mob/living/carbon/human/proc/check_defib(additional_damage = 0)
+/mob/living/carbon/human/proc/check_defib(additional_health_increase = 0)
 
 	var/datum/limb/head/head = get_limb("head")
 	if(head.limb_status & LIMB_DESTROYED)
 		return DEFIB_FAIL_DECAPITATED
 
-	if(HAS_TRAIT(src, TRAIT_UNDEFIBBABLE) && !issynth(src) || suiciding) // Synthetics who manually DNR should be caught by the next check.
+	if(HAS_TRAIT(src, TRAIT_UNDEFIBBABLE))
 		return DEFIB_FAIL_BRAINDEAD
 
-	if(!has_working_organs() && !(species.species_flags & ROBOTIC_LIMBS)) // Ya organs dpmt wprl
+	if(!has_working_organs()) // Robots and Synthetics don't have hearts or brains.
```

Comment:
avoid comments when you don't need them btw, it gets messy and you've got a lot in this pr.